suite-native/module-device-onboarding/src/components/SwipeableWalkthrough/SwipeableWalkthroughStep.tsx

```diff
@@ -0,0 +1,128 @@
+    callout: ReactNode;
+    onButtonPress: () => void;
+    title: ReactNode;
+    children: ReactNode;
+    description?: ReactNode;
+};
+
+const SPRING_ANIMATION_CONFIG = {
+    damping: 24,
+    mass: 1,
+    stiffness: 115,
+};
+const SCREEN_HEIGHT = getScreenHeight();
+
+export const SwipeableWalkthroughStep = ({
+    isStepActive,
+    animationDirection,
+    callout,
+    title,
+    description,
+    children,
+    onButtonPress,
+}: SwipeableWalkthroughStepProps) => {
+    // we need to transform animationDirection to a reanimated shared value
+    // so it can be handled inside of the worklet animation builders.
+    const reanimatedAnimationDirection = useDerivedValue<SwipeableWalkthroughAnimationDirection>(
+        () => animationDirection,
+        [animationDirection],
+    );
+
+    const customEntering = (targetValues: EntryAnimationsValues) => {
+        'worklet';
+
+        let initialOriginY;
+        switch (reanimatedAnimationDirection.value) {
+            case undefined:
+                initialOriginY = targetValues.targetOriginY;
+                break;
+            case 'forward':
+                initialOriginY = SCREEN_HEIGHT;
+                break;
+            case 'backward':
+                initialOriginY = -SCREEN_HEIGHT;
+                break;
+        }
+
+        return {
+            initialValues: {
+                originY: initialOriginY,
+            },
+            animations: {
+                originY: withSpring(targetValues.targetOriginY, SPRING_ANIMATION_CONFIG),
+            },
+        };
+    };
+
+    const customExiting = (values: ExitAnimationsValues) => {
+        'worklet';
+
+        let targetOriginY;
+        switch (reanimatedAnimationDirection.value) {
+            case undefined:
+                targetOriginY = values.currentOriginY;
+                break;
+            case 'forward':
+                targetOriginY = -SCREEN_HEIGHT;
+                break;
+            case 'backward':
+                targetOriginY = SCREEN_HEIGHT;
+                break;
+        }
+
+        return {
+            initialValues: {
+                originY: values.currentOriginY,
+            },
+            animations: {
+                originY: withSpring(targetOriginY, SPRING_ANIMATION_CONFIG),
+            },
+        };
+    };
+
+    if (!isStepActive) {
+        return null;
+    }
+
+    return (
+        <AnimatedBox
+            paddingTop="sp16"
+            flex={1}
+            justifyContent="space-between"
+            alignItems="center"
+            entering={customEntering}
+            exiting={customExiting}
+        >
+            <SwipeableWalkthroughStepHeader
+                callout={callout}
+                title={title}
+                description={description}
+            />
+            {children}
+
+            <IconButton
+                iconName="arrowDown"
+                colorScheme="tertiaryElevation0"
+                size="large"
```

Comment:
nit: the size of the button seems to be bigger in the implemented [version](https://github.com/trezor/trezor-suite/blob/b941bedda4f373ef73dfb7cdffa8d310e80871ed/suite-native/atoms/src/Button/IconButton.tsx#L35) compared to the design
![Screenshot 2025-03-28 at 12 35 29](https://github.com/user-attachments/assets/5a2f0b66-3675-4ea5-bb26-b7266d315447)

Reply:
Nit: Not sure if onboarding will be possible on iOS as well with the BT introduction, but the button position is off.
![Screenshot 2025-03-28 at 12 35 42](https://github.com/user-attachments/assets/4d380010-ec1c-4881-902f-dbfb0f1c5f3d)